frontend/src/pages/TermPlanner/ImportPlannerMenu/ImportPlannerMenu.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { useRef } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Button, Radio } from 'antd';
+import axios from 'axios';
+import { Course } from 'types/api';
+import { PlannerCourse, PlannerYear, Term } from 'types/planner';
+import openNotification from 'utils/openNotification';
+import type { RootState } from 'config/store';
+import {
+  addToUnplanned,
+  moveCourse,
+  setUnplannedCourseToTerm,
+  toggleSummer,
+  updateDegreeLength,
+  updateStartYear,
+} from 'reducers/plannerSlice';
+import CS from '../common/styles';
+import S from './styles';
+
+const ImportPlannerMenu = () => {
+  const exportFormats = ['json'];
+  const planner = useSelector((state: RootState) => state.planner);
+  const inputRef = useRef<HTMLDivElement>(null) as React.MutableRefObject<HTMLInputElement>;
+  const dispatch = useDispatch();
+
+  const array: string[] = [];
+
+  const download = async () => {
+    inputRef.current.click();
+  };
+
+  const uploadedJSONFile = async (e: React.ChangeEvent<HTMLInputElement>) => {
+    planner.years.forEach((year) => {
+      Object.values(year).forEach((termKey) => {
+        termKey.forEach((code) => {
+          array.push(code);
+        });
+      });
+    });
+    if (e.target.files !== null) {
+      if (e.target.files[0].type !== 'application/json') {
+        openNotification({
+          type: 'error',
+          message: 'Import file needs to be JSON.',
+          description: 'The uploaded file is not of type JSON.',
+        });
+        return;
+      }
+      const reader = new FileReader();
```

Comment:
Hmm i dont particularly like this - can't we just read the file synchronously to clean up the code? Otherwise we allow the user to fuck with us and give us race conditions.